src/workers/postUpdated.ts

```diff
@@ -194,101 +205,115 @@ type UpdatePostProps = {
   content_type: PostType;
 };
 const updatePost = async ({
+  logger,
   entityManager,
   data,
   id,
   mergedKeywords,
   questions,
   content_type = PostType.Article,
 }: UpdatePostProps) => {
-  const postType = contentTypeFromPostType[content_type];
-  const databasePost = await entityManager
-    .getRepository(postType)
-    .findOneBy({ id });
-
-  if (data?.origin === PostOrigin.Squad) {
-    data.sourceId = UNKNOWN_SOURCE;
-  }
-
-  if (
-    !databasePost ||
-    databasePost.metadataChangedAt.toISOString() >=
-      data.metadataChangedAt.toISOString()
-  ) {
-    return null;
-  }
+  const meter = opentelemetry.metrics.getMeter('api-bg');
+  return runInSpan('updatePost', async () => {
+    const postType = contentTypeFromPostType[content_type];
+    const databasePost = await entityManager
+      .getRepository(postType)
+      .findOneBy({ id });
+
+    if (data?.origin === PostOrigin.Squad) {
+      data.sourceId = UNKNOWN_SOURCE;
+    }
 
-  const title = data?.title || databasePost.title;
-  const updateBecameVisible =
-    content_type === PostType.Freeform
-      ? databasePost.visible
-      : !databasePost.visible && !!title?.length;
-
-  data.id = databasePost.id;
-  data.title = title;
-  data.visible = updateBecameVisible;
-  data.visibleAt = updateBecameVisible
-    ? databasePost.visibleAt ?? data.metadataChangedAt
-    : null;
-  data.sourceId = data.sourceId || databasePost.sourceId;
-
-  if (content_type in allowedFieldsMapping) {
-    const allowedFields = [
-      'id',
-      'visible',
-      'visibleAt',
-      'flags',
-      'yggdrasilId',
-      ...allowedFieldsMapping[content_type],
-    ];
-
-    Object.keys(data).forEach((key) => {
-      if (allowedFields.indexOf(key) === -1) {
-        delete data[key];
-      }
-    });
-  }
+    if (
+      !databasePost ||
+      databasePost.metadataChangedAt.toISOString() >=
+        data.metadataChangedAt.toISOString()
+    ) {
+      meter
+        .createCounter('post_update_date_conflict', {
+          description:
+            'How many posts were not updated because the database entry is newer than the received update',
+        })
+        .add(1);
+      logger.info(
+        { data },
+        'post not updated: database entry is newer than received update',
+      );
+      return null;
```

Comment:
```suggestion
      );
      span.setStatus({
      	code: opentelemetry.SpanStatusCode.ERROR,
      	message: 'post not updated: database entry is newer than received update',
      })
      return null;
```

Reply:
Could add this too, for a little bit of information and set the span to be errored